src/overlays/actors/ovl_En_River_Sound/z_en_river_sound.c

```diff
@@ -27,15 +30,208 @@ const ActorInit En_River_Sound_InitVars = {
     (ActorFunc)EnRiverSound_Update,
     (ActorFunc)EnRiverSound_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_River_Sound/EnRiverSound_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_River_Sound/EnRiverSound_Destroy.s")
+void EnRiverSound_Init(Actor* thisx, GlobalContext* globalCtx) {
+    /**
+     * Params format:
+     *     0xPPBB
+     *     P: Sence path index
+     *     B: Behavior
+    **/
+
+    THIS->b_14C = 0;
```

Comment:
Same as what I said later with using `EnRiverSound* this = THIS` and usage of `thisx`